sway-core/src/semantic_analysis/cei_pattern_analysis.rs

```diff
@@ -462,7 +462,10 @@ fn effects_of_expression(expr: &ty::TyExpression) -> HashSet<Effect> {
         | AbiName(_) => HashSet::new(),
         // this type of assignment only mutates local variables and not storage
         Reassignment(reassgn) => effects_of_expression(&reassgn.rhs),
-        StorageAccess(_) => HashSet::from([Effect::StorageRead]),
+        StorageAccess(_) => match look_up_type_id(expr.return_type) {
+            crate::TypeInfo::Struct { .. } => HashSet::new(),
```

Comment:
As discussed in the morning, this seems too liberal. I would have expected an extra check here to make sure that the struct has no fields (i.e. `fields.is_empty()`). And also, maybe empty enums should be checked as well because I suspect they don't generate any storage opcodes either.

Reply:
Thanks! Added a fix for this. I added a check for empty enums as well, although, one should not be able to construct a value of an empty enum, so I cannot test this check, but it would not hurt in the analysis :)